src/api/store/graph.py

```diff
@@ -87,6 +87,55 @@ def graph_actions_completed(self, context: Context, args) -> (Graph, MassEnergiz
       return None, CustomMassenergizeError(e)
 
 
+  def graph_actions_completed_by_team(self, context: Context, args) -> (Graph, MassEnergizeAPIError):
+    try:
+
+      team_id = args.get('team_id', None)
+
+      if not team_id:
+        return None, CustomMassenergizeError("Missing team_id field")
+
+      team: Team = Team.objects.get(id=team_id)
+
+      if not team:
+        return None, InvalidResourceError()
+
+      if context.is_prod and not context.user_is_admin():
+        if not team.is_published:
+          return None, CustomMassenergizeError("Content Not Available Yet")
+  
+      members = TeamMember.objects.filter(team=team).all()
+
+      completed_action_rels = []
+      for member in members:
+        completed_action_rels.extend(member.user.useractionrel_set.filter(status="DONE").all())
+
+      categories = TagCollection.objects.get(name="Category").tag_set.all()
+
+      prefetch_related_objects(completed_action_rels, "action__tags")
```

Comment:
i think you can combine line 113 and 115 into something like 
```
TagCollection.objects.prefetch_related_objects("action__tags").tag_set.all()
```

Reply:
I can't figure out code that would combine these lines; I don't think it's possible to call `prefetch_related` on the `TagCollection` model, because `action__tags` (which is what I want to pre-fetch) is a field of `UserActionRel`, not `TagCollection`.